src/VisualStudio/Core/Def/Implementation/DesignerAttribute/DesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -75,7 +76,8 @@ public async Task AnalyzeDocumentAsync(Document document, SyntaxNode bodyOpt, In
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            if (!document.Project.Solution.Workspace.Options.GetOption(InternalFeatureOnOffOptions.DesignerAttributes))
+            if (!document.Project.Solution.Workspace.Options.GetOption(InternalFeatureOnOffOptions.DesignerAttributes) ||
+                ServiceFeatureOnOffOptions.IsPowerSaveModeEnabled(document.Project))
```

Comment:
what happens to the user then?  they're just SOL?  they want something to be designable, and it just doesn't work?  How about we still at least analyze the file whne open/edited.  we just don't proactively scan them all.

Reply:
Tagging @sharwell @heejaechang for this too. We felt it would be acceptable in power save mode for designers to not open automatically, and the code view is the default. I think Heejae did suggest that we can do what you recommended - just scan attributes in current file. I should be able to make that change.

Reply:
i think a bit of a problem is that it's *really* going to be difficult for users to relate what they perceive as "broken behavior" to this option. 

Consider a totally realistic scenario:

1. i see this option and go "oh great!  i've wanted VS to do less work".
2. i turn it on and really like it.  i use vs for a couple fo weeks happily.
3. i then get around to needing to use the designer.  it totally doesn't work because the type doesn't seem designable.  I have no idea waht's going on and i think VS is totally broken.
4. i basically can't "kick" vs out of this state.  my only option (which i have to discover as well) is to go turn the option off.

THe problem here is that we've disconnected the option with the impact it may have on so many disparate parts of the UI.  Furthermore, this is no longer "pay for play".  there's no way for me to state that i'm willing to "pay".

In a case like designers, we could make you "pay" by having to at least open or edit the doc.  That would at least be a way to kick things out of the state its in.

For something like todo: we could make it have something in the todo UI to explain why it's not working.

etc. etc.

--

In other words, i strongly support this feature.  But basically each feature/module that is updated has to think through the user scenario around it and what's going to be an acceptable model for that interaction.  I think this will fall into a few general buckets.  But once we identify them, we're golden.

Reply:
@CyrusNajmabadi In power save mode, forms open in code view by default since it's more efficient. Opening designer view will require the user to explicitly click "View Designer".

Reply:
Can you click 'view designer' if the DesignerAttributer has not been detected/reported for that file?

If that's the case, I'm ok with this.  It's just not something i'm familiar with the workings of.  A screenshot of this in action @mavasani would be helpful.  Specifically, ensure that we do not have a registered designer attribute (i.e. comment it out), then enable the power-saving-feature (then maybe restart vs?) then add the attribute back in.  We shoudl see the file does not look designable.  If we can still open the designer though in a fairly simple/intuitive fashion, i would be ok with that.